pkg/v1/tarball/layer.go

```diff
@@ -90,11 +102,26 @@ func (l *layer) MediaType() (types.MediaType, error) {
 // LayerOption applies options to layer
 type LayerOption func(*layer)
 
+// WithCompression is a functional option for overriding the default
+// compression algorithm used for compressing uncompressed tarballs.
+// Also updates the mediaType to "application/vnd.oci.image.layer.v1.tar+zstd"
+// if zstd compression is selected.
+func WithCompression(compression LayerCompression) LayerOption {
+	return func(l *layer) {
+		if compression == ZStd {
+			l.mediaType = types.OCILayerZStd
```

Comment:
This conflicts a bit with `layer.WithMediaType`. I might just drop the `else` here? We could also warn or error or document this fact.

Reply:
I removed the mediaType override for both zstd and gzip in order to be consistent. If `WithCompression(compression.ZStd)` is used, then the user should also set `WithMediaType(`types.OCILayerZStd)`. I've mentioned that in the updated documentation for `WithCompression`. When the layer is created via `LayerFromOpener`, the method will check whether the layer's `mediaType` matches its `compression` and print a warning if not. What do you think about this approach?